internal/config/system.go

```diff
@@ -14,45 +16,98 @@ type System struct {
 
 	ModelServers ModelServers `json:"modelServers" validate:"required"`
 
-	ResourceProfiles map[string]ResourceProfile `json:"resourceProfiles"`
+	ResourceProfiles map[string]ResourceProfile `json:"resourceProfiles" validate:"required"`
 
 	Messaging Messaging `json:"messaging"`
 
-	// MetricsAddr is the address the metric endpoint binds to. Default is ":8080".
-	MetricsAddr string `json:"metricsAddr"`
+	// MetricsAddr is the address the metric endpoint binds to.
+	// Defaults to ":8080"
+	MetricsAddr string `json:"metricsAddr" validate:"required"`
 
-	// HealthAddr is the address the health probe endpoint binds to. Default is ":8081".
-	HealthAddress string `json:"healthAddress"`
+	// HealthAddr is the address the health probe endpoint binds to.
+	// Defaults to ":8081"
+	HealthAddress string `json:"healthAddress" validate:"required"`
 
 	// AllowPodAddressOverride will allow the pod address to be overridden by the Model objects. This is useful for development purposes.
 	AllowPodAddressOverride bool `json:"allowPodAddressOverride"`
+
+	Autoscaling Autoscaling `json:"autoscaling" validate:"required"`
 }
 
 func (s *System) DefaultAndValidate() error {
-	if s.Messaging.ErrorMaxBackoff.Duration == 0 {
-		s.Messaging.ErrorMaxBackoff.Duration = time.Minute
-	}
 	if s.MetricsAddr == "" {
 		s.MetricsAddr = ":8080"
 	}
 	if s.HealthAddress == "" {
 		s.HealthAddress = ":8081"
 	}
+
 	for i := range s.Messaging.Streams {
 		if s.Messaging.Streams[i].MaxHandlers == 0 {
 			s.Messaging.Streams[i].MaxHandlers = 1
 		}
 	}
+
+	if _, ok := s.Autoscaling.Profiles["default"]; !ok {
+		return errors.New("default autoscaling profile is required")
+	}
+
+	if s.Autoscaling.Interval.Duration == 0 {
+		s.Autoscaling.Interval.Duration = 10 * time.Second
+	}
+	if s.Autoscaling.TimeWindow.Duration == 0 {
+		s.Autoscaling.TimeWindow.Duration = 10 * time.Minute
+	}
+
+	for name, profile := range s.Autoscaling.Profiles {
+		// NOTE: Keep in sync with defaults in api pkg.
+		if profile.MaxReplicas == 0 {
+			profile.MaxReplicas = 3
```

Comment:
```suggestion
			profile.MaxReplicas = 100
```

This doesn't match the docs where we say default is 100.

Reply:
Updated to set default value to 3. I am concerned that if we set default max replicas to 100 that teams will only notice it in production. Thoughts?

Reply:
I think both is fine. 3 is good too as long as we're consistent throughout.